src/topology/locally_constant/basic.lean

```diff
@@ -222,8 +223,26 @@ coe_injective (funext h)
 theorem ext_iff {f g : locally_constant X Y} : f = g ↔ ∀ x, f x = g x :=
 ⟨λ h x, h ▸ rfl, λ h, ext h⟩
 
-protected lemma continuous [topological_space Y] (f : locally_constant X Y) : continuous f :=
-f.is_locally_constant.continuous
+section codomain_topological_space
+
+variables [topological_space Y] (f : locally_constant X Y)
+
+protected lemma continuous : continuous f := f.is_locally_constant.continuous
+
+/-- We can turn a locally-constant function into a bundled `continuous_map`. -/
+def to_continuous_map : C(X, Y) := ⟨f, f.continuous⟩
+
+instance : has_coe (locally_constant X Y) C(X, Y) := ⟨to_continuous_map⟩
+
+@[simp] lemma to_continuous_map_eq_coe : f.to_continuous_map = f := rfl
+
+@[simp] lemma coe_to_continuous_map : ((f : C(X, Y)) : X → Y) = (f : X → Y) := rfl
+
+lemma to_continuous_map_injective :
+  function.injective (to_continuous_map : locally_constant X Y → C(X, Y)) :=
+λ _ _ h, by { ext x, exact continuous_map.congr_fun h x, }
```

Comment:
This might golf to something like
```suggestion
λ _ _ h, ext (continuous_map.congr_fun h)
```
(untested)

Reply:
Fancy!